### PR TITLE
Update permissions, google provider version and added google_project_iam_member usage.

### PR DESCRIPTION
**Pull request by sharadkesarwani**
Update permissions, google provider version and added google_project_iam_member usage.

# Jira Ticket
Ref: https://spotinst.atlassian.net/browse/SPOTAUT-19555

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -45,12 +45,10 @@ resource "google_project_iam_binding" "spot-account-iam" {
 }
 
 # Authoritative for a given role. Updates the IAM policy to grant a role to a list of members. Other roles within the IAM policy for the project are preserved.
```
Comment (anuragsharma-123) [inline main.tf]: I guess, this message is no longer valid.

Comment (sharadkesarwani) [reply main.tf]: Updated message.

File: main.tf [hunk 2]
```diff
@@ -44,13 +44,11 @@ resource "google_project_iam_binding" "spot-account-iam" {
...
+resource "google_project_iam_member" "service-account-user-iam" {
     project = var.project
     role    = "roles/iam.serviceAccountUser"
-    members = [
-        google_service_account.spotserviceaccount.member
-    ]
+    member  = google_service_account.spotserviceaccount.member
 }
 # Link a Spot account to a GCP Cloud account.
 resource "spotinst_credentials_gcp" "gcp_connect" {
```
Comment (anuragsharma-123) [inline main.tf]: rename the value to connect_gcp

Comment (sharadkesarwani) [reply main.tf]: updated to connect_gcp

**Review by anuragsharma-123 — APPROVED** [review]

**Pull request merged**